txpool/txpool.go

```diff
@@ -192,576 +200,280 @@ func NewTxPool(
+// Start runs the pool's main loop in the background.
+// On each request received, the appropriate handler
+// is invoked in a separate goroutine.
+func (p *TxPool) Start() error {
+	go func() {
 
-// pruneAccountTx is a helper method for making sure the account specific queue
-// doesn't have any stale transactions with an invalid nonce [NOT Thread-safe]
-func (a *accountQueueWrapper) pruneAccountTx(
-	pruneCallback func(txn *types.Transaction), // Callback for additional prune logic
-) {
-	for {
-		// Check if the nonce is lower than what the TxPool is expecting
-		lowestNonceTx := a.accountQueue.txs.Peek()
-
-		if lowestNonceTx == nil || // There is nothing in the account specific queue
-			lowestNonceTx.Nonce >= a.accountQueue.nextNonce { // The lowest nonce tx is valid
-			// All good
-			break
+		// check if shutdown was called
+		select {
+		case <-p.shutdownCh:
+			return
+		default:
 		}
 
-		// Remove the transaction
-		a.accountQueue.logger.Debug(
-			fmt.Sprintf(
-				"Dropping txn [%s] due to a low nonce [%d < %d]",
-				lowestNonceTx.Hash.String(),
-				lowestNonceTx.Nonce,
-				a.accountQueue.nextNonce,
-			),
-		)
-		a.accountQueue.txs.Pop()
-
-		pruneCallback(lowestNonceTx)
-	}
-}
-
-// GetNonce returns the next nonce for the account
-// -> Returns the value from the TxPool if the account is initialized in-memory
-// -> Returns the value from the world state otherwise
-func (t *TxPool) GetNonce(addr types.Address) uint64 {
-	// Grab the account queue lock
-	wrapper := t.lockAccountQueue(addr, true)
-	defer wrapper.unlock()
-
-	return wrapper.accountQueue.nextNonce
-}
-
-// GetCapacity returns the current number of slots occupied and the max slot limit
-func (t *TxPool) GetCapacity() (uint64, uint64) {
-	return t.gauge.getHeight(), t.gauge.limit
-}
-
-// NumAccountTxs Returns the number of transactions in the account specific queue
-func (t *TxPool) NumAccountTxs(address types.Address) int {
-	wrapper := t.lockAccountQueue(address, false)
-	defer wrapper.unlock()
+		// handle requests
+		for {
+			select {
+			case req := <-p.enqueueReqCh:
+				go p.handleEnqueueRequest(req)
+			case req := <-p.promoteReqCh:
+				go p.handlePromoteRequest(req)
+			}
+		}
+	}()
 
-	return len(wrapper.accountQueue.txs)
+	return nil
 }
 
-func (t *TxPool) AddSigner(s signer) {
-	t.signer = s
+// Stops the pool's main loop.
+func (p *TxPool) Stop() {
+	p.shutdownCh <- struct{}{}
 }
 
-func (t *TxPool) handleGossipTxn(obj interface{}) {
-	if !t.sealing {
-		return
-	}
-
-	raw := obj.(*proto.Txn)
-	txn := new(types.Transaction)
-	if err := txn.UnmarshalRLP(raw.Raw.Value); err != nil {
-		t.logger.Error("failed to decode broadcasted txn", "err", err)
-	} else {
-		if err := t.addImpl(OriginGossip, txn); err != nil {
-			if !errors.Is(err, ErrAlreadyKnown) {
-				t.logger.Error("failed to add broadcasted txn", "err", err)
-			}
-		}
-	}
+// Sets the signer the pool will use
+// to check a transaction's signature.
+func (p *TxPool) AddSigner(s signer) {
```

Comment:
Add is a verb used when we're adding something to an array, it should be `SetSigner` in this case

Reply:
[Renamed](https://github.com/0xPolygon/polygon-sdk/pull/287/commits/034e1c7dec054cece5fa24dad0deb98f77404b77)